plugins/slack/server/auth/slack.ts

```diff
@@ -167,95 +179,105 @@ if (env.SLACK_CLIENT_ID && env.SLACK_CLIENT_SECRET) {
+                collectionId,
+                events: ["documents.update", "documents.publish"],
+                settings: {
+                  url: data.incoming_webhook.url,
+                  channel: data.incoming_webhook.channel,
+                  channelId: data.incoming_webhook.channel_id,
+                },
+              },
+              { transaction }
+            );
+          });
+          break;
+        }
 
-      // this code block accounts for the root domain being unable to
-      // access authentication for subdomains. We must forward to the
-      // appropriate subdomain to complete the oauth flow
-      if (!user) {
-        if (collectionId) {
-          try {
-            const collection = await Collection.findByPk(collectionId, {
-              rejectOnEmpty: true,
-            });
-            const team = await Team.findByPk(collection.teamId, {
-              rejectOnEmpty: true,
-            });
-            return redirectOnClient(
-              ctx,
-              SlackUtils.postUrl({
-                baseUrl: team.url,
-                params: ctx.request.querystring,
-              })
+        case IntegrationType.Command: {
+          authorize(user, "update", user.team);
+
+          // validation middleware ensures that code is non-null at this point
+          const data = await Slack.oauthAccess(code!, SlackUtils.connectUrl());
+
+          await sequelize.transaction(async (transaction) => {
+            const authentication = await IntegrationAuthentication.create(
+              {
+                service: IntegrationService.Slack,
+                userId: user.id,
+                teamId: user.teamId,
+                token: data.access_token,
+                scopes: data.scope.split(","),
+              },
+              { transaction }
             );
-          } catch (err) {
-            return ctx.redirect(SlackUtils.errorUrl("unauthenticated"));
-          }
-        } else {
-          return ctx.redirect(SlackUtils.errorUrl("unauthenticated"));
+            await Integration.create(
+              {
+                service: IntegrationService.Slack,
+                type: IntegrationType.Command,
+                userId: user.id,
+                teamId: user.teamId,
+                authenticationId: authentication.id,
+                settings: {
+                  serviceTeamId: data.team_id,
+                },
+              },
+              { transaction }
+            );
+          });
+          break;
         }
-      }
 
-      // validation middleware ensures that code is non-null at this point
-      const data = await Slack.oauthAccess(code!, SlackUtils.postUrl());
-      const authentication = await IntegrationAuthentication.create({
-        service: IntegrationService.Slack,
-        userId: user.id,
-        teamId: user.teamId,
-        token: data.access_token,
-        scopes: data.scope.split(","),
-      });
+        case IntegrationType.LinkedAccount: {
```

Comment:
New handler creates an integration with no auth, as auth isn't needed for anything